enumerable_methods.rb

```diff
@@ -0,0 +1,174 @@
+module Enumerable
+  # start of my_each
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    until i >= length
+      yield self[i]
+      i += 1
+    end
+    self
+  end
+  # end of my_each
+
+  # start of my_each_with_index
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    until i >= length
+      yield self[i], i
+      i += 1
+    end
+    self
+  end
+  # end of my_each_with_index
+
+  # my_select
+  def my_select
+    return to_enum unless block_given?
+
+    new_array = []
+    i = 0
+
+    until i >= length
+      new_array << self[i] if yield self[i]
+      i += 1
+    end
+    new_array
+  end
+  # end of my_select
+
+  # my_all?
+  def my_all?
+    my_each do |item|
+      return false unless yield item
+    end
+    true
+  end
+  # end of my_all?
+
+  # my_any?
+  def my_any?
+    my_each do |item|
+      return true if yield item
+    end
+    false
+  end
+  # end of my_any?
+
+  # my_none?
+  def my_none?
+    my_each do |item|
+      return false if yield item
+    end
+    true
+  end
+  # end of my_none?
+
+  # my_count
+  def my_count(arg = nil)
+    count = 0
+
+    if block_given?
+      my_each do |ele|
+        count += 1 if yield(ele) == true
+      end
+
+    elsif arg.nil?
+      my_each do |_ele|
+        count += 1
+      end
+
+    else
+      my_each do |ele|
+        count += 1 if ele == arg
+      end
+    end
+    count
+  end
+  # end of my_count
+
+  # start of my_map
+
+  def my_map(_proc = nil)
+    new_array = []
+    my_each do |item|
+      return to_enum unless block_given?
+
+      new_array << yield(item)
+    end
+    new_array
+  end
+
+  # end of my_map
+
+  # start of my_inject
+  def my_inject(initial = nil)
+    acc = if initial.nil?
+            0
+          else
+            initial
+          end
+
+    my_each do |i|
+      acc = yield(acc, i)
+    end
+    acc
+  end
```

Comment:
- [ ] should have working `my_inject` **(covered by automated tests)** identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

####  my_inject

- [ ] combines all elements of enum by applying a binary operation, specified by a block
```ruby
(1..5).my_inject(4) { |prod, n| prod * n } == (1..5).inject(4) { |prod, n| prod * n } => 480
```
- Yours is retrieving a NameError
![image](https://user-images.githubusercontent.com/52209067/78304308-aad63600-74fb-11ea-80c5-8f4dff16b3b4.png)

- Real inject
![image](https://user-images.githubusercontent.com/52209067/77698409-2ffaa180-6f76-11ea-91d3-9778aa560411.png)

- [ ] When a symbol is provided as an argument, my_inject should combine the elements of the enum using that symbol as a named method.
```ruby
[2, 5, 6].my_inject(:+) == [2, 5, 6].inject(:+) => 13
```
- Yours is retrieving a LocalJumpError
![image](https://user-images.githubusercontent.com/52209067/78304375-d35e3000-74fb-11ea-9679-051c26752069.png)

- Real inject
![image](https://user-images.githubusercontent.com/52209067/77698534-6a643e80-6f76-11ea-9d0e-ebb0f02da179.png)